packages/components/file-button/tests/file-button.test.tsx

```diff
@@ -74,4 +75,114 @@ describe("<FileButton />", () => {
     await user.click(fileButton)
     expect(onClickMock).toHaveBeenCalledTimes(0)
   })
+
+  test("should handle file selection correctly", async () => {
+    const handleFileChangeMock = vi.fn()
+
+    const TestComponent = () => {
+      const [files, setFiles] = useState<File[] | null>(null)
+
+      const handleFileChange = (newFiles: File[] | undefined) => {
+        setFiles(newFiles || null)
+        handleFileChangeMock(newFiles)
+      }
+
+      return (
+        <VStack>
+          <Text data-testid="file-count">
+            files: {files ? files.length : 0}
+          </Text>
+          <FileButton onChange={handleFileChange} multiple>
+            Upload
+          </FileButton>
+        </VStack>
+      )
+    }
+
+    render(<TestComponent />)
+
+    const uploadButton = await screen.findByRole("button", { name: /Upload/i })
+    const fileCount = screen.getByTestId("file-count")
+    const fileInput = uploadButton.previousSibling as HTMLInputElement
+
+    expect(fileCount).toHaveTextContent("files: 0")
+    expect(handleFileChangeMock).not.toHaveBeenCalled()
+
+    const file1 = new File(["test1"], "test1.txt", { type: "text/plain" })
+    fireEvent.change(fileInput, { target: { files: [file1] } })
+
+    await waitFor(() => {
+      expect(fileCount).toHaveTextContent("files: 1")
+      expect(handleFileChangeMock).toHaveBeenCalledWith([file1])
+    })
+
+    const file2 = new File(["test2"], "test2.txt", { type: "text/plain" })
+    fireEvent.change(fileInput, { target: { files: [file1, file2] } })
+
+    await waitFor(() => {
+      expect(fileCount).toHaveTextContent("files: 2")
+      expect(handleFileChangeMock).toHaveBeenCalledWith([file1, file2])
+    })
+
+    fireEvent.change(fileInput, { target: { files: [] } })
+
+    await waitFor(() => {
+      expect(fileCount).toHaveTextContent("files: 0")
+      expect(handleFileChangeMock).toHaveBeenCalledWith([])
+    })
```

Comment:
@hironokobari
To achieve 100% coverage, ensure there is a test case for when `files` is null.
This will verify that the function correctly handles the null case by returning undefined.

```diff
- fireEvent.change(fileInput, { target: { files: [] } })
+ fireEvent.change(fileInput, { target: { files: null } })

    await waitFor(() => {
      expect(fileCount).toHaveTextContent("files: 0")
-     expect(handleFileChangeMock).toHaveBeenCalledWith([])
+     expect(handleFileChangeMock).toHaveBeenCalledWith(undefined)
    })
```